compiler/optimizer/VPHandlers.cpp

```diff
@@ -2162,7 +2163,7 @@ TR::Node *constrainIaload(OMR::ValuePropagation *vp, TR::Node *node)
           && base->getClassType()
           && base->getClassType()->asFixedClass()
           && base->getClass()
-          && TR::Compiler->cls.isClassInitialized(vp->comp(), base->getClass()))
+          && (!needInitializedCheck || TR::Compiler->cls.isClassInitialized(vp->comp(), base->getClass())))
          {
          if (symRef == vp->comp()->getSymRefTab()->findJavaLangClassFromClassSymbolRef())
```

Comment:
This is'nt directly related to your commit but while reviewing this change I saw another snippet of code guarded by if (symRef == vp->comp()->getSymRefTab()->findJavaLangClassFromClassSymbolRef()) further down around line 2214 as part of the same if-then-else if construct that starts at line 2168. If you agree the code around line 2214 is unnecessary, please delete it to clean up the code.